src/build/app/mod.rs

```diff
@@ -724,26 +724,30 @@ impl<'help> App<'help> {
     /// **NOTE:** The template system is by design very simple. Therefore, the
     /// tags have to be written in the lowercase and without spacing.
     ///
-    /// Tags arg given inside curly brackets.
+    /// Tags are given inside curly brackets.
     ///
     /// Valid tags are:
     ///
-    ///   * `{bin}`         - Binary name.
-    ///   * `{version}`     - Version number.
-    ///   * `{author}`      - Author information.
-    ///   * `{about}`       - General description (from [`App::about`])
-    ///   * `{usage}`       - Automatically generated or given usage string.
-    ///   * `{all-args}`    - Help for all arguments (options, flags, positionals arguments,
-    ///                       and subcommands) including titles.
-    ///   * `{unified}`     - Unified help for options and flags. Note, you must *also* set
-    ///                       [`AppSettings::UnifiedHelpMessage`] to fully merge both options and
-    ///                       flags, otherwise the ordering is "best effort"
-    ///   * `{flags}`       - Help for flags.
-    ///   * `{options}`     - Help for options.
-    ///   * `{positionals}` - Help for positionals arguments.
-    ///   * `{subcommands}` - Help for subcommands.
-    ///   * `{after-help}`  - Help from [`App::after_help`]
-    ///   * `{before-help}`  - Help from [`App::before_help`]
+    ///   * `{bin}`                 - Binary name.
+    ///   * `{version}`             - Version number.
+    ///   * `{author}`              - Author information.
+    ///   * `{author-with-newline}` - Author followed by `\n`.
+    ///   * `{about}`               - Description of the program.
+    ///   * `{about-with-newline}`  - Description followed by `\n`.
```

Comment:
I don't think this and the `author-with-newline` should exist. The templates can have additional `\n` with the normal `about` and `author`.

Reply:
Trouble is, the `\n` has to be conditional. If about/author is not set then there is no output, otherwise it's the value plus a newline.

Originally I had multiple templates and chose between them, but quickly realized there's a combinatorial explosion with things like before/after help, etc. And building up a template dynamically seemed like it defeated the point.

The "right" way to do this is to extend the template language to support conditionals (e.g. `{if author}{author}\n{/if}`), but keeping the language simple is a great goal so I didn't go there.

Thoughts?

Reply:
I suspected something like that being the reason. Looks like we will have to be satisfied with this for now.